apps/job-core/src/amqp_to_redis.js

```diff
@@ -0,0 +1,76 @@
+import redis from 'redis'
+import { v4 as uuid } from 'uuid'
+import { promisify } from 'util'
+
+const sleep = promisify(setTimeout)
+
+class AMQPToRedis {
+  constructor(redis_url) {
+    this.client = redis.createClient({ url: redis_url })
+    this.lpop = promisify(this.client.lpop).bind(this.client)
+    this.quit = promisify(this.client.quit).bind(this.client)
+    this.closed = false
+    this.ack_callbacks = {}
+    this.exec_message = promisify((content, message_callback, callback) => {
+      const id = uuid()
+      this.ack_callbacks[id] = callback
+      message_callback({id: id, content: content})
+    }).bind(this)
+  }
+
+  queueName(queue) {
+    return 'queue_' + queue
+  }
+
+  ack(message) {
+    const callback = this.ack_callbacks[message.id]
+    if (callback) {
+      this.ack_callbacks[message.id] = null
```

Comment:
Does this every get `delete`-ed?  Or will this array just fill up with nulls until it crashes from OOM?

Reply:
Can we simply:
```suggestion
      delete this.ack_callbacks[message.id]
```

Reply:
No risk of OOM, because the app is written for AMQP

Reply:
I'm not sure I understand.  Maybe the maintainers will be able to shed some light.